features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/voicemessages/composer/VoiceMessageComposerPlayer.kt

```diff
@@ -81,13 +107,29 @@ class VoiceMessageComposerPlayer @Inject constructor(
         /**
          * The elapsed time of this player in milliseconds.
          */
-        val currentPosition: Long,
+        val curPositionMs: Long,
+        /**
+         * The duration of this player in milliseconds.
+         */
+        val durationMs: Long,
     ) {
         companion object {
             val NotPlaying = State(
                 isPlaying = false,
-                currentPosition = 0L,
+                curPositionMs = 0L,
+                durationMs = 0L,
             )
         }
+
+        val curPositionPercent = if (durationMs > 0 && durationMs >= curPositionMs) {
+            curPositionMs.toFloat() / durationMs.toFloat()
+        } else {
+            0f
+        }
     }
 }
+
+private data class Media(
+    val path: String,
+    val mimeType: String,
```

Comment:
Do we really need `mimeType` here? It's always `audio/ogg` by spec. Can't it be hardcoded?